### ISSUE TITLE
CodeMirror: error in renderer encodeSuggestions

### ISSUE THREAD

**Issue by sberequek** [primefaces-extensions/primefaces-extensions.github.com#648]
Hi,

I'm trying to use codeMirror with suggestions but I always get error when the component renderer suggestions. I add the information on the code and the error below. Can someone give me some solutions? Is it a bug?

controller method:

```java
public List<String> completeSuggestions(CompleteEvent event) {
      List<String> suggestions = new ArrayList<>();
      suggestions.add("first");
      suggestions.add("second");
      return suggestions;
}
```

code jsf:
```xml
<pe:codeMirror id="bodyValue" 
             value="#{myController.contentValue}" 
             keyMap="default" mode="javascript" theme="mdn-like" 
             widgetVar="templateVar" 
             completeMethod="#{myController.completeSuggestions}" 
             lineNumbers="true" class="Wid50" 
             extraKeys="{ 'Ctrl-Space': function(cm) { PF('templateVar').complete(); }}"/>
```

log:
`SEVERE [javax.enterprise.resource.webcontainer.jsf.context] (default task-1) java.lang.NullPointerException
	at org.primefaces.extensions.component.codemirror.CodeMirrorRenderer.encodeSuggestions(CodeMirrorRenderer.java:188)
	at org.primefaces.extensions.component.codemirror.CodeMirrorRenderer.encodeEnd(CodeMirrorRenderer.java:84)
	at javax.faces.component.UIComponentBase.encodeEnd(UIComponentBase.java:949)
	at javax.faces.component.UIComponent.encodeAll(UIComponent.java:1912)
	at com.sun.faces.context.PartialViewContextImpl$PhaseAwareVisitCallback.visit(PartialViewContextImpl.java:639)
	at com.sun.faces.component.visit.PartialVisitContext.invokeVisitCallback(PartialVisitContext.java:183)
	at javax.faces.component.UIComponent.visitTree(UIComponent.java:1736)
	at javax.faces.component.UIComponent.visitTree(UIComponent.java:1747)
	at javax.faces.component.UIComponent.visitTree(UIComponent.java:1747)
	at javax.faces.component.UIForm.visitTree(UIForm.java:386)
	at javax.faces.component.UIComponent.visitTree(UIComponent.java:1747)
	at javax.faces.component.UIComponent.visitTree(UIComponent.java:1747)
	at com.sun.faces.context.PartialViewContextImpl.processComponents(PartialViewContextImpl.java:424)
	at com.sun.faces.context.PartialViewContextImpl.processPartial(PartialViewContextImpl.java:343)
	at org.primefaces.context.PrimePartialViewContext.processPartial(PrimePartialViewContext.java:57)
	at javax.faces.component.UIViewRoot.encodeChildren(UIViewRoot.java:1124)
	at javax.faces.component.UIComponent.encodeAll(UIComponent.java:1905)
	at com.sun.faces.application.view.FaceletViewHandlingStrategy.renderView(FaceletViewHandlingStrategy.java:465)
	at com.sun.faces.application.view.MultiViewHandler.renderView(MultiViewHandler.java:194)
	at javax.faces.application.ViewHandlerWrapper.renderView(ViewHandlerWrapper.java:151)
	at javax.faces.application.ViewHandlerWrapper.renderView(ViewHandlerWrapper.java:151)
	at javax.faces.application.ViewHandlerWrapper.renderView(ViewHandlerWrapper.java:151)
	at com.sun.faces.lifecycle.RenderResponsePhase.execute(RenderResponsePhase.java:126)
	at com.sun.faces.lifecycle.Phase.doPhase(Phase.java:100)
	at com.sun.faces.lifecycle.LifecycleImpl.render(LifecycleImpl.java:223)
	at javax.faces.webapp.FacesServlet.service(FacesServlet.java:671)
	at io.undertow.servlet.handlers.ServletHandler.handleRequest(ServletHandler.java:74)
	at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:129)
	at io.opentracing.contrib.jaxrs2.server.SpanFinishingFilter.doFilter(SpanFinishingFilter.java:55)
	at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61)
	at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)
	at io.undertow.servlet.handlers.FilterHandler.handleRequest(FilterHandler.java:84)
	at io.undertow.servlet.handlers.security.ServletSecurityRoleHandler.handleRequest(ServletSecurityRoleHandler.java:62)
	at io.undertow.servlet.handlers.ServletChain$1.handleRequest(ServletChain.java:68)
	at io.undertow.servlet.handlers.ServletDispatchingHandler.handleRequest(ServletDispatchingHandler.java:36)
	at org.wildfly.elytron.web.undertow.server.ElytronRunAsHandler.lambda$handleRequest$1(ElytronRunAsHandler.java:68)
	at org.wildfly.security.auth.server.FlexibleIdentityAssociation.runAsFunctionEx(FlexibleIdentityAssociation.java:103)
	at org.wildfly.security.auth.server.Scoped.runAsFunctionEx(Scoped.java:161)
	at org.wildfly.security.auth.server.Scoped.runAs(Scoped.java:73)
	at org.wildfly.elytron.web.undertow.server.ElytronRunAsHandler.handleRequest(ElytronRunAsHandler.java:67)
	at io.undertow.servlet.handlers.security.SSLInformationAssociationHandler.handleRequest(SSLInformationAssociationHandler.java:132)
	at io.undertow.servlet.handlers.security.ServletAuthenticationCallHandler.handleRequest(ServletAuthenticationCallHandler.java:57)
	at io.undertow.server.handlers.DisableCacheHandler.handleRequest(DisableCacheHandler.java:33)
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
	at io.undertow.security.handlers.AuthenticationConstraintHandler.handleRequest(AuthenticationConstraintHandler.java:53)
	at io.undertow.security.handlers.AbstractConfidentialityHandler.handleRequest(AbstractConfidentialityHandler.java:46)
	at io.undertow.servlet.handlers.security.ServletConfidentialityConstraintHandler.handleRequest(ServletConfidentialityConstraintHandler.java:64)
	at io.undertow.servlet.handlers.security.ServletSecurityConstraintHandler.handleRequest(ServletSecurityConstraintHandler.java:59)
	at io.undertow.security.handlers.AbstractSecurityContextAssociationHandler.handleRequest(AbstractSecurityContextAssociationHandler.java:43)
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
	at org.wildfly.extension.undertow.security.jacc.JACCContextIdHandler.handleRequest(JACCContextIdHandler.java:61)
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
	at org.wildfly.extension.undertow.deployment.GlobalRequestControllerHandler.handleRequest(GlobalRequestControllerHandler.java:68)
	at org.keycloak.adapters.elytron.KeycloakServletExtension.lambda$null$0(KeycloakServletExtension.java:39)
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
	at io.undertow.servlet.handlers.ServletInitialHandler.handleFirstRequest(ServletInitialHandler.java:292)
	at io.undertow.servlet.handlers.ServletInitialHandler.access$100(ServletInitialHandler.java:81)
	at io.undertow.servlet.handlers.ServletInitialHandler$2.call(ServletInitialHandler.java:138)
	at io.undertow.servlet.handlers.ServletInitialHandler$2.call(ServletInitialHandler.java:135)
	at io.undertow.servlet.core.ServletRequestContextThreadSetupAction$1.call(ServletRequestContextThreadSetupAction.java:48)
	at io.undertow.servlet.core.ContextClassLoaderSetupAction$1.call(ContextClassLoaderSetupAction.java:43)
	at org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1502)
	at org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1502)
	at org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1502)
	at org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1502)
	at org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1502)
	at io.undertow.servlet.handlers.ServletInitialHandler.dispatchRequest(ServletInitialHandler.java:272)
	at io.undertow.servlet.handlers.ServletInitialHandler.access$000(ServletInitialHandler.java:81)
	at io.undertow.servlet.handlers.ServletInitialHandler$1.handleRequest(ServletInitialHandler.java:104)
	at io.undertow.server.Connectors.executeRootHandler(Connectors.java:360)
	at io.undertow.server.HttpServerExchange$1.run(HttpServerExchange.java:830)
	at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
	at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:1985)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1487)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1378)
	at java.lang.Thread.run(Thread.java:748)`

**Comment by melloware** [comment]
1. What version of PF and PF Extensions are you using?

2. Did you follow the example here which is working: https://www.primefaces.org/showcase-ext/sections/codeMirror/jsCodeCompletion.jsf

**Comment by sberequek** [comment]
In my project use PF version 6.2 and PF Extensions version 6.2.10 . The example with jsCodeCompletion work but for my use case it is not enough.

**Comment by melloware** [comment]
Hmmm what is your `PF('templateVar').complete();`.  doing in the extra keys section?  what does that code do?

**Comment by sberequek** [comment]
I have done this configuration as it was in the example:
https://www.primefaces.org/showcase-ext/sections/codeMirror/codeCompletion.jsf

**Comment by melloware** [comment]
OK can you clone this project and make a small reproducible sample for me?  https://github.com/primefaces/primefaces-test

**Comment by sberequek** [comment]
...using the primefaces-test project I did not find the same problem :(

[primefaces-test.zip](https://github.com/primefaces-extensions/primefaces-extensions.github.com/files/2841434/primefaces-test.zip)

**Comment by melloware** [comment]
OK then something must be going on in your code or possible version conflict?

**Comment by sberequek** [comment]
....until now I have not noticed conflicts. In my project using WildFly 14.0.1 and these dependencies:
```
        <dependency>
            <groupId>javax</groupId>
            <artifactId>javaee-web-api</artifactId>
            <version>8.0</version>
            <scope>provided</scope>
        </dependency>        
        <dependency>
            <groupId>com.sun.faces</groupId>
            <artifactId>jsf-api</artifactId>
            <version>2.2.18</version>
        </dependency>
        <dependency>
            <groupId>com.sun.faces</groupId>
            <artifactId>jsf-impl</artifactId>
            <version>2.2.18</version>
        </dependency>
        <dependency>
            <groupId>org.primefaces</groupId>
            <artifactId>primefaces</artifactId>
            <version>6.2</version>
        </dependency>
        <dependency>
            <groupId>org.primefaces.extensions</groupId>
            <artifactId>primefaces-extensions</artifactId>
            <version>6.2.10</version>
        </dependency>
       <dependency>
            <groupId>org.keycloak</groupId>
            <artifactId>keycloak-core</artifactId>
            <version>4.8.3.Final</version>
        </dependency>
        <dependency>
            <groupId>org.keycloak</groupId>
            <artifactId>keycloak-adapter-core</artifactId>
            <version>4.8.3.Final</version>
        </dependency>
        <dependency>
            <groupId>org.keycloak</groupId>
            <artifactId>keycloak-servlet-filter-adapter</artifactId>
            <version>4.8.3.Final</version>
        </dependency>
```

**Comment by melloware** [comment]
Those all look good.  That is really weird.   Can you try in the test project with `6\. mvn clean -P mojarra22 jetty:run-exploded`  That runs it with Mojarra 2.2 which is the same version WildFly is using.

Other than that i am stumped?

**Comment by sberequek** [comment]
... I think I understand it's a bug. In my xhtml code I have the form with prependId = "false" and a tag h: inputHidden id = "id", renaming the inputHidden id everything works. To give you a test case, I attach the modified test project to the page that generates an error.
[primefaces-test.zip](https://github.com/primefaces-extensions/primefaces-extensions.github.com/files/2842061/primefaces-test.zip)

**Comment by melloware** [comment]
Ahh so you had it generically with "id" with prepend=false.     Thanks for the test case!

**Comment by sberequek** [comment]
Unfortunately :( . You are welcome

**Comment by melloware** [comment]
Great test case.  Fixed it with this commit for 7.0: https://github.com/primefaces-extensions/core/commit/3512a6ac8dd22ac8c3588765c6374c23dbeb44b3
